File: src/7.整数反转.py
#
# @lc app=leetcode.cn id=7 lang=python3
#
# [7] 整数反转
#

# @lc code=start
class Solution:
    def reverse(self, x: int) -> int:
        if x == 0:
            return 0
        s = ""
        if x < 0:
            x = -x
            s += "-"
        while x%10 == 0:
            x //= 10
        while x != 0:
            s, x = s+str(x%10), x//10
        x = eval(s)
        if x > 2**31-1 or x < -2**31:
            x = 0
        return x
# @lc code=end

if __name__ == "__main__":
    s = Solution()
    print(s.reverse(123))
    print(s.reverse(-123))
    print(s.reverse(-120))
    print(s.reverse(-0))

File: src/7.整数反转.go
package main

/*
 * @lc app=leetcode.cn id=7 lang=golang
 *
 * [7] 整数反转
 */
// math

import (
	"math"
)

// @lc code=start
func reverse(x int) int {
	if x == 0 {
		return 0
	}
	flag, r := 1, 0
	if x < 0 {
		x, flag = -x, -1
	}
	for ; x%10 == 0; x /= 10 {
	}
	for ; x != 0; x /= 10 {
		r = r*10 + x%10
	}
	r *= flag
	if r > math.MaxInt32 || r < math.MinInt32 {
		return 0
	}
	return r
}

// @lc code=end
// func main() {
// 	fmt.Println(reverse(-123))
// }
